### DISCUSSION TITLE
Is there a set of AV1 instructions that work with NVenCC

### DISCUSSION THREAD

**Discussion by chowderhead** [staxrip/staxrip#1522] [Q&A]
very basic question....THERES MANY X264 AND X265 PARAMS THAT DONT WORK WITH NVenCC was gonna try and see what av1 is like but where do you put the additional flags.  I assumed maybe you stack them at the end in the CUSTOM field but thats not very functional.  any suggestions.  Otherwise AOMEnc seems the most robust but im sure its very slow without a graphics chip behind it.

**Comment by Dendraspis** [comment]
I don't get it. What's wrong with NVEncC when using AV1? As far as I can tell it works great.

**Comment by chowderhead** [comment]
i havent goten a single file to finish or even start in AV1.  Can you tell me settings that work as i havent found any that do yet.
I tried these settings with nvencc but it fails at start every time.

**Comment by Dendraspis** [comment]
What else do you need?

<img width=333 src=https://github.com/user-attachments/assets/a46ea2b5-9820-43e9-9fbd-833ecf360227>

**Comment by chowderhead** [comment]
yea i pretty much figured that part out but it fails with the canned settings for NvencC   Ihavent gotten a singlr AV1 rip to work or finish.  It starts and then suddenly dwindles down to 1.5 fps.  and just sits there doing barly anything.

**Comment by chowderhead** [comment]
honestly i dont know what im doing wrong.  I finallly dwindled down to a test AV1 encode of a small x264 file and it never finished after 8 hours of running  heres what it was doing.  my NvencC params were --qvbr 28 --codec av1 .  thats it and it still never finished.  I never really payed attention to AV1 and I guess its just as well because it seems nonfunctional.

do you have any params lists that i could try and get an actual encode to finish that you suggest??  I had hoped to get 5-10 files as a test conversion from x264 but no luck even getting 1.

any suggestions you can give for an NvencC AV1 encode would be appreciated as i appear to be at a standstill trying to use this format.  using hevc was never this difficult by comparison even in the very early stages.

thanks.

![image](https://github.com/user-attachments/assets/ec40848f-57e3-4cc8-9b41-c61bfcf05bdf)

**Reply by Dendraspis** [reply]
Can you share that small test file with me?

**Comment by chowderhead** [comment]
heres one file i tested thats a better riginal.  slice off a small piece of it if you wish but its a better sample than he the other file i was testing.  both behave identically with quick start and then slow decline down to 1-2fps and then just freezes at somee spot along the way.  other members on my discord server found documents about NVENC being hardware controlled on the GPU DIE but they also found stuff saying that ONLY RTX4080 and after can do AV1 hardware accelerated encode.  otherwise its software cpu side with no speed.   its equally strange that when running just av1 video in potplayer it taps GPU for 15% even though the rtx2070 has no AV1 silicon.  so how does that work?  

heres the sample i was using.  thanks.
https://drive.google.com/file/d/1u28i6jtDZWHmatel2Tdfh5RsSQI7NbzK/view?usp=sharing

**Comment by Dendraspis** [comment]
Ahh, didn't realize it earlier:
It seems you have ignored my questions/screenshot and you use `ffmpeg` instead of `NVEncC`, even you have mentioned `NVEncC`....

**Comment by chowderhead** [comment]
HUH?  what makes you think im using ffmpeg?  I clearly selected NVencC AV! from the staxrip menu just like the picture.  I didnt add or change anything and the result's as ive shown start quick but trail off to 1.47fps and then eventually just dies around 20-30% finish.

maybe your confused by the file link I sent you.  This file on GDrive is the SOURCE FILE I use to convert from 264 to AV1.  I didnt send you the output file which of course never finishes so i cant share it with you,  

I truly dont know why you think im using ffmpeg.
![image](https://github.com/user-attachments/assets/2bb6ebfd-74e4-4fcf-8c38-8d902fbc1a4d)

**Reply by Dendraspis** [reply]
Because of the output/progress. It does not look like NVEncC.

Can you provide the log file so I can see what happened and how to reproduce it?

**Comment by chowderhead** [comment]
I updated my stax to 2.42.1 from 2.36 and will rerun the file very deliberately picking the NVencC AV1 menu and reproduce the error if possible on the most recent version.
thanks for all your effort to finally address the audio tracks situatin.  I know it was a huge rework but its so MUCH better its hard to describe how much time it saves when workig on 100+ episode series with >2 audio tracks.  thank you so much!

ok with version 2.42.1 i got a much different response and an error that explains it all,
______________________________________________________________________________________
GPU #0 (NVIDIA GeForce RTX 2070) does not support AV1 encoding.
Unknown erro occurred during checking GPU.


Video encoding returned exit code: 1 (0x1)
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

It's unclear what this exit code means, in case it's
a Windows system error then it possibly means:

# for hex 0x1 / decimal 1
  STATUS_WAIT_1                                                  ntstatus.h   
  ERROR_INVALID_FUNCTION                                         winerror.h   
# Incorrect function.
# as an HRESULT: Severity: SUCCESS (0), FACILITY_NULL (0x0), Code 0x1
# for hex 0x1 / decimal 1
  ERROR_INVALID_FUNCTION                                         winerror.h   
# Incorrect function.


Start:    8:24:32 AM
End:      8:24:32 AM
Duration: 00:00:00
___________________________________________________________________________________________
SOMETHING IN 2.42.1 IS DIFFERENT.  ERRORS THAT MEAN SMETHING.

**Reply by Dendraspis** [reply]
Oh boy .... that's the reason why you should provide as many information as possible, for example via log file!

So yes, that's the reason....

And I hope you have started with new settings and templates to avoid side effects!

**Comment by chowderhead** [comment]
I had no idea you need RTX 4070 80 or90 to do AV1 encoding.  i spent hours at AV1 for dummies.  NEVER read that my 2070 was incopatible.
it just blows up and gives a REAL WORLD ERROR THAT CLEARS ALL THIS UP.  Until 2.42.1 ive never seen that error I pasted above until this update.

**Comment by chowderhead** [comment]
I oresu,e AV1 CPU encoding works I guess
do you have a set of suggested basic settings for the AOM Ripping engine,
Just curious,  I get a lot of errors  when doing CPU encode as well  geez what a pain.
its so eassy to crash these rip engines for AV1.

I found the performance of AVIF as a jpeg contender interesting as well,

**Reply by Dendraspis** [reply]
The easiest and well balanced AV1 CPU encoder is `SvtAv1EncApp`, that's my recommendation. It's default values are also a very good starting point.

**Reply by chowderhead** [reply]
thanks,  appreciate the tip.  Interesting that the giant conglomerate AOM encoder is not the most balanced.  i wanna understand  superblocks and how AV1 uses this better.

ALSO, just fyi but you nay recall 2.36 had an issue for me that rips would fail w strange uninformative memory or stack errors 3-4x more often if i put ANY of staxrip/NVencC encodes into the background by simply minimizing the window.  NVencC was definitely part of issue but debug only REAL answer was to NOT use NVencC.  Thats not helpful.  (hohummmmm)...   Basically Annoying but impossible to debug.  I tried and just ran in circles.  I had a test folder of tiny 75MB files to rip in background as tests but no matter what I changed (except not using NVencC), NO LUCK . Well as fate goes, 2.42.1 no longer does this.  _ just ripped 20 files minimized window with no issue so_ thats a real blessing.  thanks for fixing something you weren't trying to fix and many many thanks for the audio upgrade fix that at least for me IS VERY HELPFUL.!! especially on batches with mixed audio tracks. 😎

**Reply by Dendraspis** [reply]
> thanks for fixing something you weren't trying to fix

Not sure if I wasn't aware of this issue. But to be honest I don't even remember how `v2.42.1` looked like and worked.  😅 

> many many thanks for the audio upgrade

You're welcome. 😊

**Comment by chowderhead** [comment]
> And I hope you have started with new settings and templates to avoid side effects!

I actually have a block of templates I cant do without so i just launch the old staxrip 2,36 and 2,42 at same time and recreate each one from the old visually recreating to avoid any issues.   Kinda a pain but only real way to keep them and avoid issues.  ya know, ....Theres really A LOT of info in each template you dont realize how much till ya gotta remake 20 or so of them takes a good 30-40 minutes but well worth it. 
 hahaha  😃😃

**Reply by Dendraspis** [reply]
> i just launch the old staxrip 2,36 and 2,42 at same time and recreate each one from the old visually recreating to avoid any issues.

That's the recommended way to do it when you upgrade to the next major version, for example `v2.42.x` to `v2.43.x`.

> Kinda a pain but only real way to keep them and avoid issues.

Definitely!

> remake 20 or so of them takes a good 30-40 minutes but well worth it.

Indeed, that's a lot. But like you already mentioned:  Better to avoid strange and unwanted side effects.  👍
